toolkit/tools/imagecustomizer/docs/clone-rpm-repo.md

```diff
@@ -0,0 +1,177 @@
+   | 2.0                 | x86_64 | https://packages.microsoft.com/cbl-mariner/2.0/prod/base/x86_64/  |
+   | 2.0                 | ARM64  | https://packages.microsoft.com/cbl-mariner/2.0/prod/base/aarch64/ |
+   | 3.0                 | x86_64 | https://packages.microsoft.com/azurelinux/3.0/prod/base/x86_64/   |
+   | 3.0                 | ARM64  | https://packages.microsoft.com/azurelinux/3.0/prod/base/aarch64/  |
+
+   For example:
+
+   ```bash
+   REPO_URL="https://packages.microsoft.com/azurelinux/3.0/prod/base/x86_64/"
+   ```
+
+3. Clone PMC.
+
+   If you want clone all of PMC, then run:
+
+   ```bash
+   dnf reposync --repofrompath "azurelinux,$REPO_URL" --repo azurelinux --newest-only
+   ```
+
+   If you want to only clone a subset of packages (and their dependencies), then run:
+
+   ```bash
+   PACKAGE_LIST="vim nano"
+   dnf download --repofrompath "azurelinux,$REPO_URL" --repo azurelinux --resolve --alldeps --destdir azurelinux $PACKAGE_LIST
+   ```
+
+   This will download the RPMs into a directory named `azurelinux`.
+
+4. Cache the downloaded RPMs somewhere.
+
+5. Use cached RPMs with the image customizer tool.
+
+   ```bash
+    sudo ./imagecustomizer \
+      --build-dir ./build \
+      --image-file <base-image-file> \
+      --output-image-file ./out/image.vhdx \
+      --output-image-format vhdx \
+      --config-file <config-file> \
+      --disable-base-image-rpm-repos \
+      --rpm-source <rpms-dir>
+   ```
+
+   where:
+
+   - `<base-image-file>`: The base image file.
+   - `<config-file>`: The image customizer config file.
+   - `<rpms-dir>`: The local directory that contains the downloaded RPMs.
+
+## Hosting a cloned repo
+
+It may be desirable to host the downloaded RPMs in a common location so that it can be
+used by both builds and developers.
+
+An RPM server is simply a HTTP server that hosts static files. There is no dynamic
+content. So, pretty much any HTTP server application or provider can be used. The files
+served by the HTTP server are the RPM files themselves and a few metadata files that
+document what RPMs are available.
+
+Example RPM server using httpd/apache2:
+
+1. Install prerequisites:
+
+   Azure Linux 2.0 and 3.0:
+
+   ```bash
+   sudo tdnf -y install createrepo_c httpd
+   sudo systemctl enable --now httpd
+   ```
+
+   Ubuntu 22.04:
+
+   ```bash
+   sudo apt update -y
+   sudo apt install -y createrepo-c apache2
+   ````
+
+2. Download the cached RPMs to a local directory.
+
+3. Create the metadata files:
+
+   ```bash
+   createrepo_c --compatibility --update <rpms-dir>
+   ```
+
+   where:
+
+   - `<rpms-dir>`: The directory you downloaded the RPMs to.
+
+4. Move the RPMs directory:
+
+   ```bash
+   sudo mkdir -p /var/www
+   sudo mv -T <rpms-dir> /var/www/rpms
+   ```
+
+5. Configure the HTTP server:
+
+   Azure Linux 2.0 and 3.0:
+
+   ```bash
+   sudo sed -i 's|"/etc/httpd/htdocs"|"/var/www/rpms"|' /etc/httpd/conf/httpd.conf
```

Comment:
Will we also need to add iptable rules to allow traffic through port 80?

Reply:
I think Azure Linux now defaults to firewalls that are open by default. So, I didn't need to do this.